tests/test_ecommerce.py

```diff
@@ -371,16 +372,42 @@ def test_metadata():
         "title": "E-commerce",
         "description": "Template for spiders that extract product data from e-commerce websites.",
         "param_schema": {
+            "groups": [
+                {
+                    "description": (
+                        "Input data that determines the start URLs of the crawl."
+                    ),
+                    "id": "inputs",
+                    "title": "Inputs",
+                    "widget": "exclusive",
+                },
+            ],
             "properties": {
                 "url": {
+                    "default": "",
                     "description": (
                         "Initial URL for the crawl. Enter the full URL including http(s), "
                         "you can copy and paste it from your browser. Example: https://toscrape.com/"
                     ),
+                    "exclusiveRequired": True,
+                    "group": "inputs",
                     "pattern": r"^https?://[^:/\s]+(:\d{1,5})?(/[^\s]*)*(#[^\s]*)?$",
                     "title": "URL",
                     "type": "string",
                 },
+                "seed_url": {
```

Comment:
> Specially since whether or not the URLs are seed URLs would determine whether or not we try automatic extraction on them.

Ahhh this made me realize that the PO would have `response: AnyResponse` _(i.e. for the first request's response)_, and if `response` is a `BrowserResponse`, we don't have the response headers to check. Thus, it'd be tricky to do the implicit approach via `"Content-Type": "text/plain"` header checks. 🙅‍♂️

> Or implement some client-side syntax to indicate that a URL is a seed URL,

I would avoid some non-standard syntax that needs to be learned by the user here.

-------

One last thing to discuss and sort out: _What about the use case of a user trying to crawl multiple seed URLs?_

For example, the user has two or more pre-defined textfiles which are a bunch of websites per country:

- https://somesource.com/fr-websites.txt
  - https://fr.example-1.com
  - https://fr.example-2.com
  - https://fr.example-3.com
  - https://fr.example-4.com
- https://somesource.com/de-websites.txt
  - https://de.example-1.com
  - https://de.example-2.com
  - https://de.example-3.com
- https://somesource.com/es-websites.txt
  - ...
- ...

And say, they want to make a virtual spider that crawls EMEA websites.

In order to support it, they can re-use the existing textfiles they have to crawl them by passing them to the `urls` param:

- https://somesource.com/fr-websites.txt
- https://somesource.com/de-websites.txt
- https://somesource.com/es-websites.txt
- ...

Moreover, there's a `boolean` flag that they need to turn on to treat such URLs as seeds.

Reply:
I suggested the boolean approach originally, but @VMRuiz believes that, when using seed URLs, the bast majority of cases will use a single seed URL. Which I did not realize at the beginning, but makes sense.

Reply:
Yeah, I don't think this would be a huge dealbreaker to users as well, as they can simply make a new list with the combined URLs. 

Glad all other use cases were explored!